features/kafka/producer/src/main/java/org/opennms/features/kafka/producer/OpennmsKafkaProducer.java

```diff
@@ -191,7 +199,25 @@ public boolean shouldForwardAlarm(OnmsAlarm alarm) {
         return true;
     }
 
+    private boolean isIncrementalAlarm(String reductionKey, OnmsAlarm alarm) {
+        return outstandingAlarms.containsKey(reductionKey) &&
+                alarmEqualityChecker.equalsExcludingOnFirst(protobufMapper.toAlarm(alarm).build(),
+                        outstandingAlarms.get(reductionKey));
+    }
+
+    private void recordIncrementalAlarm(String reductionKey, OnmsAlarm alarm) {
+        // Apply the excluded fields when putting to the map so we do not have to perform this calculation
+        // on each equality check
+        outstandingAlarms.put(reductionKey,
+                AlarmEqualityChecker.Exclusions.defaultExclusions(protobufMapper.toAlarm(alarm)).build());
+    }
+
     private void updateAlarm(String reductionKey, OnmsAlarm alarm) {
+        if (suppressIncrementalAlarms && isIncrementalAlarm(reductionKey, alarm)) {
```

Comment:
The `alarm` may be null here. I'd defer this check until after the filtering call to `shouldForwardAlarm`.